src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/ConversionCompletionProvider.cs

```diff
@@ -120,9 +125,16 @@ private ImmutableArray<CompletionItem> GetBuiltInEnumConversions(SemanticModel s
             // No suggestion for the other two kinds of conversions:
             // * From sbyte, byte, short, ushort, int, uint, long, ulong, char, float, double, or decimal to any enum_type.
             // * From any enum_type to any other enum_type.
-            if (container.IsEnumMember() || // SomeEnum.EnumValue.$$
+
+            // Suggest after enum members: SomeEnum.EnumValue.$$
+            var suggestBuiltInEnumConversion = container.IsEnumMember();
+
+            // Suggest for enum values, but not after the enum type (don't infer with the enum member listing)
+            suggestBuiltInEnumConversion = suggestBuiltInEnumConversion ||
                 (container.IsEnumType() &&  // someEnumVariable.$$
-                    !(semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol is ITypeSymbol typeSymbol && typeSymbol.IsEnumType()))) // but not SomeEnum.$$
+                    !(semanticModel.GetSymbolInfo(expression, cancellationToken).Symbol is ITypeSymbol typeSymbol && typeSymbol.IsEnumType())); // but not SomeEnum.$$
```

Comment:
can't understand this at all :)

Reply:
I removed this check here because what I wanted to check is also valid for all other indexers/operator/conversions:
*Don't offer suggestions on static member access.*
The unit tests were added in 53ad16fa.
The fix, which is essentially the same as what was removed here, was moved to `OperatorIndexerCompletionProviderBase` in 7370df62:

```CS
        public override async Task ProvideCompletionsAsync(CompletionContext context)
        {

            var expressionSymbolInfo = semanticModel.GetSymbolInfo(expression, cancellationToken);
            if (expressionSymbolInfo.Symbol is INamedTypeSymbol)
            {
                // Expression looks like an access to a static member. Operator, indexer and conversions are not applicable.
                return;
            }
        }
```

I later added also tests (3d344893) and a fix (3429c57c) for `nameof` context.